components/profile/AdminPanel.tsx

```diff
@@ -0,0 +1,157 @@
+import { Menu, Dropdown } from 'antd';
+import { DownOutlined } from '@ant-design/icons';
+import React from "react";
+import { Mentions } from 'antd';
+import 'antd/dist/antd.css';
+import useSWR,{mutate,trigger} from "swr";
+import fetcher from "../../lib/utils/fetcher";
+import { SERVER_BASE_URL } from "../../lib/utils/constant";
+import { Button } from 'antd';
+import TagAPI from "../../lib/api/tag";
+import { Alert } from 'antd';
+
+
+const { Option } = Mentions;
+
+const Admin = (props) => {
+   
+
+    const [value, setValue] = React.useState("Select tag")
+    const [key, setKey] = React.useState("")
+
+    const [selectedOption, setSelectedOption] = React.useState()
+    const [alert, setAlert] = React.useState(false)
+    const [alertMessage,setAlertMessage] = React.useState("")
+    const [alertType,setAlertType] = React.useState()
+    const [tag,setTag] = React.useState([
+
+    ])
+    const [tagMemberOptions,setTagMemberOptions] = React.useState([
+
+    ])
+    const tagMembers= async(key)=>{ 
+        var options =[]
+        if(key != null){
+          var res =   await TagAPI.getMembers(key)
+          var tagMembers = await res.data
+          var tagMem = []
+        if (tagMembers && tagMembers.tagFollowers.length > 0) {
+        for (var members of tagMembers.tagFollowers) {
+            tagMem.push(members.profile.username)
+            options.push(<Option value={members.profile.username}>
+                {members.profile.username}
+        </Option>)
+       
+        }
+    }
+       setTagMemberOptions(tagMem)
+     }
+     return options
+     
+    }
+    const handleClickFavorite = async (slug,selectedOption) => {
+        
+        if(slug != null && selectedOption != null && selectedOption.length != 0 && selectedOption !="@"){
+        if(tagMemberOptions && tagMemberOptions.includes(selectedOption) ){
+        mutate(
+            `${SERVER_BASE_URL}/tags/${slug}/moderator/${selectedOption}`,
+            {},
+            true
+          );
+                var response = await TagAPI.moderators(slug,selectedOption)
+                        if(response && response.status == 200){
```

Comment:
I would use the ES6 if else statement conditionals to shorten this if statement. You can make the code inside of the if and the else into separate functions and call them respectively. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator